app/frontend/src/features/communities/discussions/DiscussionPage.test.tsx

```diff
@@ -0,0 +1,113 @@
+import {
+  render,
+  screen,
+  waitForElementToBeRemoved,
+} from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { Route, Switch } from "react-router-dom";
+import { discussionBaseRoute, discussionRoute } from "routes";
+import { service } from "service";
+import discussions from "test/fixtures/discussions.json";
+import threadRes from "test/fixtures/getThreadRes.json";
+import { getHookWrapperWithClient } from "test/hookWrapper";
+import { getUser } from "test/serviceMockDefaults";
+import {
+  assertErrorAlert,
+  mockConsoleError,
+  MockedService,
+  wait,
+} from "test/utils";
+
+import { PREVIOUS_PAGE } from "../constants";
+import DiscussionPage, { CREATOR_LOADING_TEST_ID } from "./DiscussionPage";
+
+const getUserMock = service.user.getUser as MockedService<
+  typeof service.user.getUser
+>;
+const getDiscussionMock = service.discussions.getDiscussion as MockedService<
+  typeof service.discussions.getDiscussion
+>;
+const getThreadMock = service.threads.getThread as MockedService<
+  typeof service.threads.getThread
+>;
+
+function renderDiscussion() {
+  const { client, wrapper } = getHookWrapperWithClient({
+    initialRouterEntries: [
+      "/previous-page",
+      `${discussionBaseRoute}/1/what-is-there-to-do-in-amsterdam`,
+    ],
+    initialIndex: 1,
+  });
+  render(
+    <Switch>
+      <Route exact path={discussionRoute}>
+        <DiscussionPage />
+      </Route>
+      <Route exact path="/previous-page">
+        <h1 data-testid="previous-page">Previous page</h1>
+      </Route>
+    </Switch>,
+    { wrapper }
+  );
+
+  return client;
+}
+
```

Comment:
These don't cover much the replies section for now as that needs more work in the next PRs, so only a starting point